src/main/java/core/basesyntax/strategy/impl/SupplyOperation.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.Operation;
+import core.basesyntax.strategy.OperationException;
+
+public class SupplyOperation implements Operation {
+    private final StorageDao storageDao;
+
+    public SupplyOperation(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void proceed(FruitTransaction fruitTransaction) {
+        if (storageDao.get(fruitTransaction.getFruitName()) == null) {
+            throw new OperationException("Fruit " + fruitTransaction.getFruitName()
+                    + " doesn't exist");
+        }
+        Fruit fruit = storageDao.get(fruitTransaction.getFruitName());
+        fruit.setQuantity(fruit.getQuantity() + fruitTransaction.getQuantity());
```

Comment:
Make sure to update the storage with the new quantity of the fruit after you change it. The current code modifies the `fruit` object, but it doesn't persist the update back to the storage.